### PR TITLE
core&ui: fix bugs

### PR DESCRIPTION
**Pull request by Molmin**
1. 修复比赛题目详情页 **标题栏** 题目按钮无效

   例：https://hydro.ac/p/279?tid=645e688dcd521724464af308
2. 修复复制来的题目的附件的显示 (#661)
3. 修复比赛中题目提交页面、比赛讨论列表页面侧边栏不显示比赛参加状态
4. ~~修复个人主页题解数量显示~~ (reverted)

### CODE REVIEW

File: packages/ui-default/templates/partials/problem_files.html [hunk 1]
```diff
@@ -6,7 +6,8 @@
     <blockquote class="note">
       <p>{{ _('This is a copy of another problem.') }}</p>
     </blockquote>
-  {% elif _files.length %}
+  {% endif %}
+  {% if _files.length and (filetype == 'additional_file' or not reference) %}
```
Comment (undefined-moe) [inline packages/ui-default/templates/partials/problem_files.html]: please note that file upload and editing should not be allowed

File: packages/hydrooj/src/model/solution.ts [hunk 2]
```diff
@@ -72,6 +73,8 @@ class SolutionModel {
     static async vote(domainId: string, psid: ObjectId, uid: number, value: number) {
         let pssdoc = await document.getStatus(domainId, document.TYPE_PROBLEM_SOLUTION, psid, uid);
         await document.setStatus(domainId, document.TYPE_PROBLEM_SOLUTION, psid, uid, { vote: value });
+        const { owner } = await this.get(domainId, psid);
```
Comment (undefined-moe) [inline packages/hydrooj/src/model/solution.ts]: do not use `this` in static method

**Review by undefined-moe — CHANGES_REQUESTED** [review]

File: packages/hydrooj/src/model/solution.ts [hunk 3]
```diff
@@ -72,6 +73,8 @@ class SolutionModel {
     static async vote(domainId: string, psid: ObjectId, uid: number, value: number) {
         let pssdoc = await document.getStatus(domainId, document.TYPE_PROBLEM_SOLUTION, psid, uid);
         await document.setStatus(domainId, document.TYPE_PROBLEM_SOLUTION, psid, uid, { vote: value });
+        const { owner } = await SolutionModel.get(domainId, psid);
+        await domain.incUserInDomain(domainId, owner, 'nLike', (value === 1 ? 1 : 0) - (pssdoc && pssdoc.vote === 1 ? 1 : 0));
```
Comment (pandadtdyy) [inline packages/hydrooj/src/model/solution.ts]: If you want to add it, please clean it when delete a solution.

**Review by pandadtdyy — CHANGES_REQUESTED** [review]

File: packages/hydrooj/src/model/solution.ts (hunk 3 above)
Comment (Molmin) [reply packages/hydrooj/src/model/solution.ts]: Should I calculate it together with downvote to improve efficiency?

**Review by pandadtdyy — APPROVED** [review]

**Review by undefined-moe — APPROVED** [review]

**Pull request merged**